src/rgw/rgw_reshard.cc

```diff
@@ -604,23 +604,19 @@ int RGWBucketReshard::do_reshard(int num_shards,
 	  }
 	}
 
-	if (verbose) {
+	if (verbose && out) {
 	  formatter->close_section();
-	  if (out) {
-	    formatter->flush(*out);
-	  }
+	  formatter->flush(*out);
```

Comment:
ack

Reply:
Reverted & apologizing for the confusion, this hunk got staged by mistake, did not mean to add a formatter / revert commit cc27e35  
Just was reproducing the memory leak as in nautilus,mimic,luminous
For example in Nautilus it was:
https://github.com/ceph/ceph/blob/534c2f14693439413791049826ca23d1706c7a8d/src/rgw/rgw_reshard.cc#L1023-L1024